src/rgw/rgw_auth.h

```diff
@@ -430,28 +434,32 @@ class LocalApplier : public IdentityApplier {
 
 protected:
   const RGWUserInfo user_info;
-  const std::string subuser;
-
-  uint32_t get_perm_mask(const std::string& subuser_name,
-                         const RGWUserInfo &uinfo) const;
+  std::string subuser;
```

Comment:
It seems those fields have been made non-const due to `parse_subuser_info`. As they are tightly coupled, we might consider a small `struct` or even `std::pair<std::string, uint32_t>`. Then `parse_subuser_info` could return an object of such type instead of `void`.

The variant with `struct` might look like:
```cpp
  const struct subuser_t {
    std::string name;
    uint32_t perm_mask;
  } subuser;
```
Then, for example, in `parse_subuser_info`:
```cpp
return subuser_t { iter->second.name, RGW_PERM_FULL_CONTROL };
```